### ISSUE TITLE
alacritty_terminal: infinite loop when double-width character resized into single column terminal

### ISSUE THREAD

**Issue by ConradIrwin** [alacritty/alacritty#7697]
If you resize an alacritty terminal to 1 column wide while it is displaying a double-width character (in our example "🦀"), then it infinite loops and [this allocation](https://github.com/alacritty/alacritty/blob/6d37de167f3a9b7da75ee208d19cceac5f8f0867/alacritty_terminal/src/grid/resize.rs#L314) will continually increase the amount of RAM used until the process is killed by the system.

The fix we've gone with is to ensure that we always render at least 2 columns: https://github.com/zed-industries/zed/pull/7471, but filing this in case it's of interest to other embedders. (It does not affect alacritty itself which seems to have a minimum width of closer to 10 columns).

### System

We're embedding alacritty v0.21 in [Zed](https://zed.dev).

**Comment by kchibisov** [comment]
alacritty sets the minimum amount of columns to `2` as well, since the alacritty_terminal supports only minimum of `1x2` sizes.

That should probably be a part of `resize` routine in `alacritty_terminal`...

**Comment by chrisduerr** [comment]
Yeah single-column is just not supported. It doesn't really make sense considering a fullwidth character is two columns wide.

**Comment by chrisduerr** [comment]
> The fix we've gone with is to ensure that we always render at least 2 columns

Note that what Alacritty does is limiting the **terminal** to 2 columns, not the rendering. Alacritty works with a 1x1 window size last I checked.

**Comment by kchibisov** [comment]
yeah, but we limit that in alacritty itself, not `alacritty_term`, we could probably limit in `alacritty_term` instead, so you kind of not resize to 1x1.

**Comment by chrisduerr** [comment]
> yeah, but we limit that in alacritty itself, not alacritty_term, we could probably limit in alacritty_term instead, so you kind of not resize to 1x1.

Yeah that's fair, we shouldn't allow resizing below our minimum size.
